test/algo/compression/pruning/test_concrete_trace_mmdetection.py

```diff
@@ -0,0 +1,315 @@
+    if RoIAlign_solution in (1, 2):
+        from mmcv import ops as mmcv_ops
+        leaf_module_append = (mmcv_ops.RoIAlign,)
+
+    model = init_detector(config, device=device)
+
+    with torch.no_grad():
+        cfg = model.cfg
+        cfg.data.test.pipeline = replace_ImageToTensor(cfg.data.test.pipeline)
+        test_pipeline = Compose(cfg.data.test.pipeline)
+        img_data = test_pipeline(dict(img_info=dict(filename=img), img_prefix=None))['img']
+        img_tensor = collate(img_data, 1).data[0].to(device)
+
+        # init run
+        # some models need to be run 2 times before doing trace
+        model.forward_dummy(torch.rand_like(img_tensor))
+        model.forward_dummy(torch.rand_like(img_tensor))
+
+        seed = torch.seed()
+        torch.manual_seed(seed)
+        out_orig_1 = model.forward_dummy(img_tensor)
+        torch.manual_seed(seed)
+        out_orig_2 = model.forward_dummy(img_tensor)
+        assert check_equal(out_orig_1, out_orig_2), 'check_equal failure for original model'
+        del out_orig_1, out_orig_2
+
+        # torch.manual_seed(100)
+
+        if config_file == 'pvt/retinanet_pvt-l_fpn_1x_coco':
+            # to support numpy.intc
+            import torch.fx as torch_fx
+            from numpy import intc, int64
+            orig_base_types = torch_fx.proxy.base_types
+            torch_fx.proxy.base_types = (*torch_fx.proxy.base_types, intc, int64)
+
+        traced_model = concrete_trace(model, {'img': img_tensor},
+                                        use_function_patch = False, forwrad_function_name='forward_dummy',
+                                        autowrap_leaf_function = {
+            **ConcreteTracer.default_autowrap_leaf_function,
+            **leaf_function_append,
+            all:                                                                    ((), False, None),
+            min:                                                                    ((), False, None),
+            max:                                                                    ((), False, None),
+        }, autowrap_leaf_class = {
+            **ConcreteTracer.default_autowrap_leaf_class,
+            int:        ((), False),
+            reversed:   ((), False),
+            torch.Size: ((), False),
+        }, leaf_module = (
+            *leaf_module_append,
+            mmcv_cnn.bricks.wrappers.Conv2d,
+            mmcv_cnn.bricks.wrappers.Conv3d,
+            mmcv_cnn.bricks.wrappers.ConvTranspose2d,
+            mmcv_cnn.bricks.wrappers.ConvTranspose3d,
+            mmcv_cnn.bricks.wrappers.Linear,
+            mmcv_cnn.bricks.wrappers.MaxPool2d,
+            mmcv_cnn.bricks.wrappers.MaxPool3d,
+        ), fake_middle_class = (
+            mmdet_core.anchor.anchor_generator.AnchorGenerator,
+        ))
+
+        if config_file == 'pvt/retinanet_pvt-l_fpn_1x_coco':
+            torch_fx.proxy.base_types = orig_base_types
+
+        seed = torch.seed()
+        torch.manual_seed(seed)
```

Comment:
If seed is manually set, it should be restored to a random seed (e.g., based on timestamp) after this test so that other tests won't be affected.

Reply:
The seed comes from `torch.seed` which would return a random seed, So subsequent tests will not be affected.

Reply:
If seed is a random seed, why bother? You could simply omit setting seed.